src/engine/engine_wb.c

```diff
@@ -204,15 +204,15 @@ int ocf_write_wb(struct ocf_request *req)
 		 */
 		ocf_engine_map(req);
 
-		if (!req->info.eviction_error) {
+		if (!req->info.not_inserted) {
 			/* Lock request for WRITE access */
 			lock = ocf_req_trylock_wr(req);
 		}
 
 		OCF_METADATA_UNLOCK_WR(); /*- END Metadata WR access ---------*/
 	}
 
-	if (!req->info.eviction_error) {
+	if (!req->info.not_inserted) {
```

Comment:
If we have a partial hit due to to promotion policy (some request core lines not promoted to the cache) the remaining cache lines will be invalidated. Isn't this a design error?

Reply:
Yeah, I guess it looks like that from the API. The idea here is that `should_promote` takes in a full request after traversation - and it may choose to engage or not based on full request state not line-by-line scanning. For now I've come up with those cases when it should let the cache do its thing normally:
* cache is not full - no reason to filter anything, but I guess this one could be configured in runtime
* request is partially dirty
* or more generally - request is  a partial hit

Maybe we should put it down in writing just to make sure we've filled all the gaps.